### PR TITLE
fix a bug in joining empty partitioned tables

### PR DESCRIPTION
**Pull request by kovrus**
Return `NoopPlan` for `SelectStatementPlanner` if
the planned sub relation in `visitMultiSourceSelect` is
`NoopPlannedAnalyzedRelation`.

### CODE REVIEW

File: sql/src/test/java/io/crate/integrationtests/JoinIntegrationTest.java [hunk 1]
```diff
@@ -108,6 +108,24 @@ public void testInsertFromInnerJoin() throws Exception {
...
+        execute("create table bar (id long) partitioned by (id)");
+        ensureYellow();
+        execute("select * from foo f, bar b where f.id = b.id");
+        assertThat(printedTable(response.rows()), is(""));
+    }
+
+    @Test
+    public void testJoinOnEmptyTables() throws Exception {
+        execute("create table foo (id long)");
+        execute("create table bar (id long)");
```
Comment (mfussenegger) [inline sql/src/test/java/io/crate/integrationtests/JoinIntegrationTest.java]: does this even trigger the NoopPlan case?

File: sql/src/main/java/io/crate/planner/SelectStatementPlanner.java [hunk 2]
```diff
@@ -192,6 +193,9 @@ public Plan visitMultiSourceSelect(MultiSourceSelect mss, Planner.Context contex
             ConsumerContext consumerContext = new ConsumerContext(mss, context);
             // plan sub relation as if root so that it adds a mergePhase
             PlannedAnalyzedRelation plannedSubQuery = consumingPlanner.plan(mss, consumerContext);
+            if (plannedSubQuery instanceof NoopPlannedAnalyzedRelation) {
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/planner/SelectStatementPlanner.java]: maybe add a comment why there can be a NoopPlannedAnalyzedRelation here

File: sql/src/test/java/io/crate/integrationtests/JoinIntegrationTest.java (hunk 1 above)
Comment (kovrus) [inline sql/src/test/java/io/crate/integrationtests/JoinIntegrationTest.java]: removing this test

**Comment by mfussenegger** [comment]
lgtm

**Pull request merged**